BoneLib/BoneLib/HelperMethods.cs

```diff
@@ -132,13 +141,26 @@ public static byte[] GetResourceBytes(Assembly assembly, string name)
             }
             return null;
         }
-        
+
         ///<summary>
         /// Checks if an assembly is loaded from name
         /// </summary>
         public static bool CheckIfAssemblyLoaded(string name)
         {
-            return AppDomain.CurrentDomain.GetAssemblies().Any(asm => asm.GetName().Name.ToLower().Contains(name.ToLower()));
+            HashSet<Assembly> query = new HashSet<Assembly>(AppDomain.CurrentDomain.GetAssemblies());
```

Comment:
Why bother making a HashSet if you're just going to iterate it like the array it already comes out as?